libiocage/lib/JailFilter.py

```diff
@@ -0,0 +1,165 @@
+from typing import List, Union, Iterable
+import re
+import libiocage.lib.Jail
+import libiocage.lib.errors
+
+
+def match_filter(value: str, filter_string: str):
+    escaped_characters = [".", "$", "^", "(", ")", "?"]
+    for character in escaped_characters:
+        filter_string = filter_string.replace(character, f"\\{character}")
+    filter_string = filter_string.replace("*", ".*")
+    filter_string = filter_string.replace("+", ".+")
+    pattern = f"^{filter_string}$"
+    match = re.match(pattern, value)
+    return match is not None
+
+
+class Term(list):
+
+    glob_characters = ["*", "+"]
+
+    def __init__(self, key, values=list()):
+        self.key = key
+
+        if values is None:
+            raise TypeError("Values may not be empty")
+        elif isinstance(values, str):
+            data = self._split_filter_values(values)
+        else:
+            data = [values]
+
+        list.__init__(self, data)
+
+    def matches_jail(self, jail: libiocage.lib.Jail.JailGenerator) -> bool:
+        return self.matches(jail.getstring(self.key))
+
+    def matches(self, value: str) -> bool:
+        """
+        Returns True if the value matches the term
+        """
+        for filter_value in self:
+
+            if self.key == "name":
+                if self._validate_name_filter_string(filter_value) is False:
+                    raise libiocage.lib.errors.JailFilterInvalidName(
+                        filter_value,
+                        logger=self.logger
```

Comment:
The class Term has no attribute `logger`, if it needs one, we should extend its `__init__`

Reply:
there's now a patch for this #62